dlt/destinations/impl/duckdb/configuration.py

```diff
@@ -69,6 +69,12 @@ def parse_native_representation(self, native_value: Any) -> None:
             # check if database was passed as explicit connection
             import duckdb
 
+            # if in-memory mode has been requested
+            # then prepare duckdb connection so later
+            # it is configured appropriately
+            if isinstance(native_value, str) and native_value == ":memory:":
```

Comment:
I'm not sure if we want to do this here, or just amend the docs to show how to hand over and in memory instance of duckdb, but if we keep it like this, you need to also to the following:
* set the database property correctly (the way it is written now it is set to external)
* If we create the instance here, we also are the owner

Reply:
Hey @sh-rp I think this is a legitimate use case and a better UX/DX for our users.
I thought about setting `_conn_owner = True` but refrained, I will adjust this and set it.